llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -1248,9 +1248,18 @@ bool InstCombinerImpl::replaceInInstruction(Value *V, Value *Old, Value *New,
     return false;
 
   auto *I = dyn_cast<Instruction>(V);
-  if (!I || !I->hasOneUse() || !isSafeToSpeculativelyExecute(I))
+  if (!I || !I->hasOneUse() || !isSafeToSpeculativelyExecuteWithoutInstrInfo(I))
     return false;
 
+  // Special handling for replacing called operand
+  if (auto *Call = dyn_cast<CallInst>(I)) {
```

Comment:
It is hard to provide a negative test.

Reply:
I'd rather not have this special case. Can we limit isSafeToSpeculativelyExecute to only intrinsics if !UseInstrInfo? I don't think we really care about the generic case here.